recipes/dfba/meta.yaml

```diff
@@ -0,0 +1,66 @@
+{% set name = "dfba" %}
+{% set version = "0.1.8" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: fdeeecab9bfe169c091382195eab32d49545a140e14741d446b4c164a62ec24a
+
+build:
+  skip: true  # [win]
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv --no-deps
+
+requirements:
+  build:
+    - cmake
+    - {{ compiler('cxx') }}
+    - python
+  host:
+    - glpk
+    - pybind11
+    - python
+    - sundials
+    - pip
+  run:
+    - cobra
+    - depinfo
+    - {{ compiler('cxx') }}
```

Comment:
I don't think this needs a c++ compiler at *runtime*?

Reply:
Hi @dopplershift the software uses a c++ compiler to compile a shared library at runtime (that is then loaded by the main program). I discussed with @Midnighter today whether it would be important to get a runtime test implemented to ensure this works cross-platform with conda-forge (as we have done [here](https://gitlab.com/davidtourigny/dynamic-fba)).

Reply:
AH, had no idea. Yeah, it would be really good to have a test that makes sure that this compilation part functions correctly.

Reply:
Hi @dopplershift, maybe you know how to do this best. I've included a `run_test.py` file in the recipe's directory which gets executed automatically so that's working great. However, I have a problem with getting the model file (`iJR904.xml.gz`) to be mounted in the directory where the tests are run. I have tried to do this by listing it in the `source_files` but that leads to the following error. What's the best way to achieve this?

```
Traceback (most recent call last):
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 175, in <module>
    build_all(os.path.join(root_dir, "recipes"), args.arch)
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 100, in build_all
    build_folders(recipes_dir, folders, arch, channel_urls)
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 156, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/opt/conda/lib/python3.8/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 3068, in build_tree
    packages_from_this = build(metadata, stats,
  File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 2351, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 1657, in bundle_conda
    output['checksums'] = create_info_files(metadata, replacements, files, prefix=metadata.config.host_prefix)
  File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 1266, in create_info_files
    copy_test_source_files(m, join(m.config.info_dir, 'test'))
  File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 830, in copy_test_source_files
    raise RuntimeError(msg.format(pattern))
RuntimeError: Did not find any source_files for test with pattern iJR904.xml.gz
```

Reply:
Use [`files:`](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-files) rather than `source_files:`. (I had to look this up.)

Reply:
Sweet, that did it thanks!